src/angular-cesium-widgets/services/zoom-to-rectangle.service.ts

```diff
@@ -0,0 +1,243 @@
+ *
+ * @function activate -
+ * @param options
+ * {
+ *  onStart - optional - a callback that will be called when the user start drawing the rectangle
+ *  onComplete - optional - a callback that will be called when the tool zoom in
+ *  autoDisableOnZoom - optional - determines if the tool should auto disable after zoom - default: true
+ *  animationDurationInSeconds - optional - zoom animation duration in seconds - default: 0.5
+ *  borderStyle - optional - the style of the rectangle element border - default: '3px dashed #FFFFFF'
+ *  backgroundColor - optional - the background color of the rectangle element - default: 'transparent'
+ *  resetKeyCode - optional - the key code of the key that is used to reset the drawing of the rectangle - default: 27 (ESC key)
+ * }
+ * @param mapId - optional - the mapId of the map that the tool will be used in.
+ *
+ */
+
+@Injectable()
+export class ZoomToRectangleService {
+  private cameraService: CameraService;
+  private cesiumService: CesiumService;
+  constructor(
+    private mapsManager: MapsManagerService,
+    @Optional() cameraService: CameraService,
+    @Optional() cesiumService: CesiumService,
+  ) {}
+
+  init(cesiumService: CesiumService, cameraService: CameraService) {
+    this.cameraService = cameraService;
+    this.cesiumService = cesiumService;
+  }
+
+  private mapsZoomElements = new Map<string, ZoomData>();
+  private defaultOptions = {
+    animationDurationInSeconds: 0.5,
+    resetKeyCode: 27,
+    borderStyle: '2px solid rgba(0,0,0,0.5)',
+    backgroundColor: 'rgba(0,0,0,0.2)',
+    autoDisableOnZoom: true,
+  };
+
+  activate(
+    options: {
+      onStart?: (acMap?: AcMapComponent) => any;
+      onComplete?: (acMap?: AcMapComponent) => any;
+      autoDisableOnZoom?: boolean;
+      animationDurationInSeconds?: number;
+      borderStyle?: string;
+      backgroundColor?: string;
+      resetKeyCode?: number;
+    } = {},
+    mapId?: string,
+  ) {
+    if ((!this.cameraService || !this.cesiumService) && !mapId) {
+      throw new Error(`The function must receive a mapId if the service wasn't initialized`);
+    }
+    const finalOptions = Object.assign({}, this.defaultOptions, options);
+    let cameraService = this.cameraService;
+    let mapContainer;
+    let map;
+    if (this.cesiumService) {
+      mapContainer = this.cesiumService.getViewer().container;
+      map = this.cesiumService.getMap();
+    }
+    if (mapId) {
+      map = this.mapsManager.getMap(mapId);
+      if (!map) {
+        throw new Error(`Map not found with id: ${mapId}`);
+      }
+      cameraService = map.getCameraService();
+      mapContainer = map.getCesiumViewer().container;
+    }
+
+    if (!cameraService || !mapContainer) {
+      throw new Error(`The function must receive a mapId if the service wasn't initialized`);
+    }
+    this.disable(mapId);
+    const container = document.createElement('div');
+    mapContainer.style.position = 'relative';
```

Comment:
isnt it a bit dangerous changing the uesrs map css?? 
why dont use `fixed` position?

Reply:
The map container is created by us and not the user so I think its fine

Reply:
We have something similar in ac-map